bindgen/codegen/mod.rs

```diff
@@ -1138,8 +1138,56 @@ impl CodeGenerator for Type {
                             })
                             .unwrap_or(ctx.options().default_visibility);
                         let access_spec = access_specifier(visibility);
+
+                        // Collect field attributes from multiple sources for newtype tuple field
+                        let mut all_field_attributes = Vec::new();
+
+                        // 1. Get attributes from typedef annotations (if any)
+                        all_field_attributes.extend(
```

Comment:
Can we deduplicate this into its own function, or is it too annoying? Seems like something like:

```rust
fn field_attributes(type_name, type_kind, field_name, field_type_name, ctx) -> TokenStream;
```

Would be doable?

Reply:
Sure I'll take a look.